docs: New docs for table functions (#3727)
confluentinc/ksql: docs/developer-guide/syntax-reference.rst

|                        |                                                                           | function is 1-indexed. ELT is the complement to   |
|                        |                                                                           | FIELD.                                            |
+------------------------+---------------------------------------------------------------------------+---------------------------------------------------+
| ENTRIES                | ``ENTRIES(map MAP, sorted BOOLEAN)``                                      | Constructs an array of structs from the entries in|

> |                        |                                                                           | a map. Each struct has a field with name `K`      |

Reviewer: ```suggestion
|                        |                                                                           | a map. Each struct has a field with name ``K``    |
```

Author: done

> |                        |                                                                           | containing the key (this is always a String) and a|

Reviewer: ```suggestion
|                        |                                                                           | containing the key, which is always a String, and a|
```

Author: done

> |                        |                                                                           | field with name `V` holding the value.            |

Reviewer: ```suggestion
|                        |                                                                           | field named ``V``, which holds the value.                   |
```

Author: done

+------------------------+---------------------------------------------------------------------------+---------------------------------------------------+
| RANDOM                 |  ``RANDOM()``                                                             | Return a random DOUBLE value between 0.0 and 1.0. |
+------------------------+---------------------------------------------------------------------------+---------------------------------------------------+
| RANGE                  |  ``RANGE(start_inclusive, end_exclusive)``                                | Constructs an array of values between             |

> |                        |                                                                           |`start_inclusive` and `end_exclusive` (exclusive). |

Reviewer: `start_inclusive` and `end_exclusive` need double backticks, which will require reformatting the table (sadly).

Author: done

> |                        |                                                                           | Parameters can be `INT' or `BIGINT`.              |

Reviewer: ```suggestion
|                        |                                                                           | Parameters can be ``INT`` or ``BIGINT``.          |
```

Author: done

.. _ksql_key_requirements:

.. _ksql_table_functions:

===============
Table functions
===============

>+------------------------+---------------------------+------------+---------------------------------------------------------------------+
>| Function               | Example                   | Input Type | Description                                                         |
>+========================+===========================+============+=====================================================================+
>| EXPLODE                | ``EXPLODE(col1)``         | Array      | This function takes an Array and outputs one value for each of the  |

> >|                        |                           |            | elements of the array. The output values are of the type of the     |                                     |

Reviewer: ```suggestion
>|                        |                           |            | elements of the array. The output values have the same type as the  |                                     |
```

Author: done
